haystack/nodes/prompt/prompt_node.py

```diff
@@ -774,11 +776,14 @@ def prompt(self, prompt_template: Optional[Union[str, PromptTemplate]], *args, *
                 # merge any additional model kwargs
                 kwargs = {**kwargs, **self._prepare_model_kwargs()}
                 # and pass the prepared prompt and kwargs to the model
+                self.constructed_prompts.append(prompt)
+                logger.debug('Prompt being sent to LLM with prompt "{}" and kwargs "{}"'.format(prompt, kwargs))
```

Comment:
Could we change this logging statement to not use the `format` call? We have found for debugging statements to the logger that this cause unnecessary execution of the string formatting even if debugging is turned off. There are more details about how this affects performance in this PR https://github.com/deepset-ai/haystack/pull/3843

So the change I'd recommend is 
```python
logger.debug('Prompt being sent to LLM with prompt "%s" and kwargs "%s"', prompt, kwargs)
```

Reply:
Sure, I have changed that now. Thanks